dart/test/sentry_event_test.dart

```diff
@@ -292,18 +298,26 @@ void main() {
       expect(serialized['exception'], null);
     });
 
-    test('should serialize stacktrace if SentryStacktrace', () {
-      final stacktrace =
-          SentryStackTrace(frames: [SentryStackFrame(function: 'main')]);
-      final serialized = SentryEvent(stackTrace: stacktrace).toJson();
-      expect(serialized['threads']['values'].first['stacktrace'], isNotNull);
+    test('should serialize thread', () {
+      final serialized = SentryEvent(threads: [
+        SentryThread(
+          id: 0,
+          crashed: true,
+          current: true,
+          name: 'Isolate',
+        )
+      ]).toJson();
+      expect(serialized['threads']['values'], isNotNull);
     });
 
-    test('should not serialize event.stacktrace if event.exception is set', () {
+    /*
+    test(
+        'should not serialize event.threads.stacktrace if event.exception is set',
```

Comment:
I'm not quite sure what the desired behavior of this test (`should not serialize event.threads.stacktrace if event.exception is set`) and the one below it (`should not serialize stacktrace if not SentryStacktrace`) should be.

Reply:
if `attachStacktrace` is enabled, the event has a `stackTrace` set, but also a `exception`, and the `exception` also has a `stackTrace`, they are the same since they read the current stack trace from the Isolate or the stackTrace from the error, so we only set the threads.stacktrace if there's not a exception, reducing the amount of sent data

Reply:
Thanks, that clears it up

Reply:
Both test never did anything. Both checked for a `stacktrace` key in the json, but that doesn't exit. Well, that's gonna change now :D

Reply:
damn, luckily we have you :D